client/app/account/user/agreements/user-agreements.service.js

```diff
@@ -25,11 +90,24 @@ angular.module("UserAccount").service("UserAccount.services.agreements", [
                     },
                     serviceType: "aapi"
                 })
+                .then(formatList)
                 .then(getSuccessDataOrReject);
         };
 
         this.getAgreement = function (agreementId) {
-            return $http.get(`${proxyPath}/agreements/${agreementId}`).then(getSuccessDataOrReject);
+            return $http.get(`${proxyPath}/agreements/${agreementId}`)
+                .then((response) => {
```

Comment:
You should use the destructuring assignment syntax here, you use only "data"

```javascript
.then(({ data }) => {
```

Reply:
I sadly can't do that, I need to return the full object in order to chain `.then(...)` statements